bioptim/optimization/variable.py

```diff
@@ -256,12 +256,17 @@ def to_dictionaries(self, data: Union[np.array, DM]) -> list:
             offset += self.n_phase_u[p]
 
         offset = self.n_all_x + self.n_all_u
-        data_parameters["all"] = v_array[offset:]
+        scaling_offset = 0
+        data_parameters["all"] = v_array[offset:, np.newaxis] * ocp.nlp[0].p_scaling
         if len(data_parameters["all"].shape) == 1:
             data_parameters["all"] = data_parameters["all"][:, np.newaxis]
         for param in self.parameters_in_list:
-            data_parameters[param.name] = v_array[offset : offset + param.size]
+            data_parameters[param.name] = (
+                v_array[offset : offset + param.size, np.newaxis]
+                * ocp.nlp[0].p_scaling[scaling_offset : scaling_offset + param.size]
+            )
             offset += param.size
+            scaling_offset += param.size
```

Comment:
Some day, this should reads directly from "all" instead of duplicating the data (same goes for the states and controls). I think this can create a performance issue if lot of data are copied... For now lets keep it like this , but when this happens all this for loop will be removed